cpp/src/groupby/hash/groupby.cu

```diff
@@ -426,15 +426,17 @@ auto create_hash_map(table_device_view const& d_keys,
 
   using map_type = concurrent_unordered_map<size_type,
                                             size_type,
-                                            row_hasher<default_hash, keys_have_nulls>,
-                                            row_equality_comparator<keys_have_nulls>>;
+                                            row_hasher<default_hash, nullate::DYNAMIC>,
```

Comment:
I'm a bit apprehensive about switching to runtime nullability in the groupby hash code.

Reply:
> I'm a bit apprehensive about switching to runtime nullability in the groupby hash code.

I verified that groupby benchmarks were not impacted. [This one](https://github.com/rapidsai/cudf/blob/83adc4b6cf2e1081ae77c28d421c60cc2d736165/cpp/benchmarks/groupby/group_sum_benchmark.cu#L43) specifically uses the groupby hash code. I updated the benchmark to include multi-column keys table to exercise the row-element-comparator as well as the hasher.

I also added non-null tests to the hashing benchmarks to show dynamic null checking does not seem to slow down hashing specifically.